src/Rings/mpoly.jl

```diff
@@ -589,6 +589,17 @@ Fields:
   end
 
   function MPolyIdeal(B::IdealGens{T}) where T
+    if length(B) >= 1
+      oscar_assure(B)
+      R = B.gens.Ox
+      if R isa MPolyDecRing
+         if is_graded(R)
```

Comment:
This kind of check could be rewritten to just do `if is_graded(R)
`, which IMHO reads much more naturally, if only we provided methods for `is_graded`, `is_z_graded`, etc. for "regular" rings and ideals where they just return `false`.

I just saw that PR #2236 does this for `is_graded`.